test/crash/src/test/java/io/narayana/lra/arquillian/LRACoordinatorRecoveryIT.java

```diff
@@ -218,22 +186,20 @@ public void lraCoordinatorRecoveryTwoLRAs(@ArquillianResource @OperateOnDeployme
         restartContainer(LRA_COORDINATOR_CONTAINER_QUALIFIER);
 
         // Waits for a period of time longer than the timeout of the short LRA transaction
-        doWait((LRAListener.LRA_SHORT_TIMELIMIT + 1L) * 1000);
-
-        int sc = recover();
-
-        if (sc != 0) {
-            recover();
-        }
+        doWait(SHORT_TIMEOUT + 1000L);
 
         LRAStatus longStatus = getStatus(longLRA);
         LRAStatus shortStatus = getStatus(shortLRA);
 
         Assert.assertEquals("LRA with long timeout should still be active",
                 LRAStatus.Active.name(), longStatus.name());
+        if (shortStatus == LRAStatus.Cancelling) {
+            recover();
+            shortStatus = getStatus(shortLRA);
+        }
         Assert.assertTrue("LRA with short timeout should not be active",
                 shortStatus == null ||
-                        LRAStatus.Cancelled.equals(shortStatus) || LRAStatus.Cancelling.equals(shortStatus));
+                        LRAStatus.Cancelled.equals(shortStatus));
```

Comment:
@marcosgopen I am a bit curious if this change was something needed to make the test pass or whether it is just hardening of the test because it looks like before `if (shortStatus == LRAStatus.Cancelling)` then the test would have passed anyway but now we are forcing it to recover and forcing to see it be Cancelled. Was that part of the purpose of the pull to move to being sure the LRA was cancelled?

Reply:
It is a hardening of the test, it is part of the purpose of the pull: we expect the LRA to be in cancelled status in the test.

Reply:
Thank you!